### PR TITLE
Members-only

### PR DESCRIPTION
**Pull request by yahir91**
In this project, we build a blog that members can create posts, and they can see who wrote the posts but the nonmembers can not see the author.
The project has a sign-up page to create members who can make posts. The members have permission to create, edit and delete posts and see the authors of the posts.
It has an authentication build with devise.

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @team :clap: :clap: However, some requirements need to be implemented in order to get approval for this milestone. Check :arrow_down: 

- **[REQUIRED]** The application has Sign in/Sign out links present in any view of the app ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))
##### Routes
- **[REQUIRED]** It creates the application routes using `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)) 
##### Views
- **[REQUIRED]** There shoud be absolutely NO logic in views.
* [ ] View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)
  > Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
##### PostsController

- **[REQUIRED]** it uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
##### General
- **[REQUIRED]** Pull Request has a meaningful title and a brief description ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
+class PostsController < ApplicationController
+  before_action :authenticate_member!, only: %i[new create]
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_member!, except: %i[index]
```
Comment (stratospherique) [inline app/controllers/posts_controller.rb]: - authenticate_member! is repeated twice and assigned for actions other that new and create. Kindly remove the last one.

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Rails.application.routes.draw do
+  devise_for :members, controllers: { sessions: 'members/sessions', registrations: 'members/registrations' }
+  resources :posts, only: [:new, :create, :index, :show]
+  resources :members
```
Comment (stratospherique) [inline config/routes.rb:4]: - **[REQUIRED]** only routes that are used should be defined.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  # POST /posts
+  # POST /posts.json
+  def create
+    @post = Post.new(post_params)
+    @post.member_id = current_member.id
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to @post, notice: 'Post was successfully created.' }
+        format.json { render :show, status: :created, location: @post }
```
Comment (stratospherique) [inline app/controllers/posts_controller.rb]: - Rendering unused JSON data can cause performance issues to the server in a production environment. Plus, you won't need it unless you have a front-end that will consume it Kindly remove all lines responsible for rendering JSON data.

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great progress @team. :clap: However, some changes need to be applied in order to get approval for this project. Check :arrow_down: 

- **[REQUIRED]** The views still have logic in them. refer to the articles below, they should help you refactor your views
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- **[REQUIRED]** Check the code comments below, they should help you make the necessary fixes to some issues that weren't fixed.
### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by rOluochKe — APPROVED** [review]
Hello @yahir91 and @maosan132 
### Well done, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next project.
- Happy coding!

### Improvements
- [ ] Remove all the commented code
- [ ] You can use bare bootstrap to style the app that way it will look better and presentable

**Pull request merged**